tests/integration/test_target_snowflake.py

```diff
@@ -820,3 +821,17 @@ def test_flush_streams_with_intermediate_flushes_on_all_streams(self, mock_emit_
 
         # Every table should be loaded correctly
         self.assert_logical_streams_are_in_snowflake(True)
+
+    def test_record_validation(self):
+        """Test validating records"""
+        tap_lines = test_utils.get_test_tap_lines('messages-with-invalid-records.json')
+
+        # Loading invalid records when record validation enabled should fail at ...
+        self.config['validate_records'] = True
+        with assert_raises(RecordValidationException):
+            self.persist_lines_with_cache(tap_lines)
+
+        # Loading invalid records when record validation disabled should fail at load time
+        self.config['validate_records'] = False
+        with assert_raises(Exception):
```

Comment:
At load time, shouldn't the raised exception be known? `Exception` is too broad, `with assert_raises(Exception):` will also catch `RecordValidationException`.

Reply:
yes thank you, I'll change it

Reply:
fixed by bc7a35d4bfbe10405bea36e01a754dc0aa879374

Reply:
hmm test failing, I’ll fix it and come back to you with greens :) thanks